components/Button.js

```diff
@@ -9,8 +9,14 @@ const StyledButton = styled.button`
   &:hover {
     cursor: pointer;
   }
+  background-color: ${({ $buttonVariant }) =>
+    $buttonVariant === "delete" ? "red" : "#EEDC82"};
 `;
 
-export default function Button({ onClick, name }) {
-  return <StyledButton onClick={onClick}>{name}</StyledButton>;
+export default function Button({ onClick, name, $buttonVariant }) {
+  return (
+    <StyledButton $buttonVariant={$buttonVariant} onClick={onClick}>
+      {name}
+    </StyledButton>
+  );
 }
```

Comment:
The `$` prefix in styled-components is used to prevent props from being forwarded to the DOM. Without it, React might attempt to add `buttonVariant="delete"` as an HTML attribute, which is not valid.

We should use the `$` prefix only inside `StyledButton`, but not in the function parameters of `Button`.

```suggestion
export default function Button({ onClick, name, buttonVariant }) {
  return (
    <StyledButton $buttonVariant={buttonVariant} onClick={onClick}>
      {name}
    </StyledButton>
  );
}
```